### PR TITLE
Connect to SQL using Managed Identities

### PR DESCRIPTION
**Pull request by adam230594**

### CODE REVIEW

File: src/SFA.DAS.Reservations.Api/AppStart/AddDatabaseExtension.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            {
+                services.AddDbContext<ReservationsDataContext>(options => options.UseSqlServer(config.ConnectionString), ServiceLifetime.Transient);
+            }
+            else
+            {
+                services.AddSingleton(new AzureServiceTokenProvider());
+                services.AddDbContext<ReservationsDataContext>(ServiceLifetime.Transient);
+            }
+
+            services.AddTransient<IReservationsDataContext, ReservationsDataContext>(provider => provider.GetService<ReservationsDataContext>());
```
Comment (adam230594) [inline src/SFA.DAS.Reservations.Api/AppStart/AddDatabaseExtension.cs:28]: Should this be .AddScoped<>

File: src/SFA.DAS.Reservations.Api/Startup.cs [hunk 2]
```diff
@@ -78,7 +79,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton(new ReservationsApiEnvironment(Configuration["Environment"]));
             
             services.AddHealthChecks()
-                    .AddSqlServer(config.Value.ConnectionString)
+                    .AddSqlServer(config.Value.ConnectionString) //Need to use AddDbDataContext?
```
Comment (adam230594) [inline src/SFA.DAS.Reservations.Api/Startup.cs:82]: This will break.

File: src/SFA.DAS.Reservations.Data/SFA.DAS.Reservations.Data.csproj [hunk 3]
```diff
@@ -6,6 +6,8 @@
 
   <ItemGroup>
     <PackageReference Include="Elasticsearch.Net" Version="7.4.0" />
+    <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.5.0" />
+    <PackageReference Include="Microsoft.Data.SqlClient" Version="2.0.1" />
```
Comment (adam230594) [inline src/SFA.DAS.Reservations.Data/SFA.DAS.Reservations.Data.csproj:10]: I could remove this package and just use System.Data.SqlClient instead

File: src/SFA.DAS.Reservations.Api/AppStart/AddDatabaseExtension.cs (hunk 1 above)
Comment (dashton82) [reply src/SFA.DAS.Reservations.Api/AppStart/AddDatabaseExtension.cs:28]: for this project/solution yeah it can be `.AddScoped`

**Review by dashton82 — APPROVED** [review]

**Pull request closed without merging**